src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -278,9 +297,9 @@ static ucs_status_t UCS_F_ALWAYS_INLINE uct_dc_mlx5_ep_am_short_iov_inline(
                                UCT_IB_MLX5_AV_FULL_SIZE);
     UCT_DC_CHECK_RES_AND_FC(iface, ep);
     UCT_DC_MLX5_IFACE_TXQP_GET(iface, ep, txqp, txwq);
+    uct_dc_mlx5_set_dgram_seg(txwq, iface, &ep->av, uct_dc_mlx5_ep_get_grh(ep));
     uct_rc_mlx5_txqp_inline_iov_post(&iface->super, UCT_IB_QPT_DCI, txqp, txwq,
-                                     iov, iovcnt, iov_length, id, &ep->av,
-                                     uct_dc_mlx5_ep_get_grh(ep),
+                                     iov, iovcnt, iov_length, id,
```

Comment:
can we return av_size from uct_dc_mlx5_set_dgram_seg, so that uct_ib_mlx5_wqe_av_size will not have an additional branch? maybe could also remove uct_ib_mlx5_wqe_av_size

Reply:
> can we return av_size from uct_dc_mlx5_set_dgram_seg, so that uct_ib_mlx5_wqe_av_size will not have an additional branch?

ok, done

> maybe could also remove uct_ib_mlx5_wqe_av_size

it is used in UD, moved into UD .c file